neqo-crypto/src/cert.rs

```diff
@@ -44,13 +44,13 @@ fn stapled_ocsp_responses(fd: *mut PRFileDesc) -> Option<Vec<Vec<u8>>> {
     let ocsp_nss = unsafe { SSL_PeerStapledOCSPResponses(fd) };
     match NonNull::new(ocsp_nss as *mut SECItemArray) {
         Some(ocsp_ptr) => {
-            let mut ocsp_helper: Vec<Vec<u8>> = Vec::new();
-            let Ok(len) = isize::try_from(unsafe { ocsp_ptr.as_ref().len }) else {
-                qerror!("[{fd:p}] Received illegal OSCP length");
+            let Ok(len) = usize::try_from(unsafe { ocsp_ptr.as_ref().len }) else {
+                qerror!("[{fd:p}] Received illegal OCSP length");
                 return None;
             };
+            let mut ocsp_helper: Vec<Vec<u8>> = Vec::with_capacity(len);
             for idx in 0..len {
-                let itemp: *const SECItem = unsafe { ocsp_ptr.as_ref().items.offset(idx).cast() };
+                let itemp: *const SECItem = unsafe { ocsp_ptr.as_ref().items.add(idx).cast() };
```

Comment:
I wonder whether this could use `collect()` instead.  Or maybe that's just too much work to rewrite.